src/main/java/com/billingapi/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,95 @@
+package com.billingapi.exception;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.MissingPathVariableException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import java.time.LocalDateTime;
+
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+
+
+    /*Exception Handling for InvoiceNotFoundException
+     */
+    @ExceptionHandler(InvoiceNotFoundException.class)
+    public ResponseEntity<ExceptionResponse> invoiceNotFoundException(Exception ex, WebRequest request) {
+        ExceptionResponse errors = new ExceptionResponse();
+        errors.setTimestamp(LocalDateTime.now());
+        errors.setMessage(ex.getMessage());
+        errors.setStatus(HttpStatus.NOT_FOUND.value());
+
+        return new ResponseEntity<>(errors, HttpStatus.NOT_FOUND);
+
+    }
+
+/* Exception Handling for Validations
+  Using   onMethodArgumentNotValidException
+
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ResponseBody
+    ErrorModelResponse onMethodArgumentNotValidException(
+            MethodArgumentNotValidException e) {
+        ErrorModelResponse error = new ErrorModelResponse();
+        for (FieldError fieldError : e.getBindingResult().getFieldErrors()) {
+            error.getErrorMessage().add(
+                    new ValidationsError("The  " + fieldError.getField(), fieldError.getDefaultMessage()));
+        }
+        return error;
+    }
+*/
+
+
+
+    /*
+    Exception Handling for Validations to check handleHttpRequestMethodNotSupported
+
+  Using Spring boot Inbuilt ResponseEntityExceptionHandler to check Method is supported or not
+
+     */
+    @Override
+    protected ResponseEntity<Object> handleHttpRequestMethodNotSupported(HttpRequestMethodNotSupportedException ex, HttpHeaders headers, HttpStatus status, WebRequest request) {
+        return new ResponseEntity<Object>("Please change your Method type", HttpStatus.NOT_FOUND);
```

Comment:
use new ExceptionResponse(ex.getMessage(), HttpStatus.Method_Not_Allowed); object to return applicable to other plces also.